runtime/src/migrations.rs

```diff
@@ -0,0 +1,162 @@
+
+            for (acc_id, acc) in prev::Account::<Runtime>::iter() {
+                let expected_data = pallet_balances::AccountData {
+                    free: acc.data.free,
+                    reserved: acc.data.reserved,
+                    frozen: acc.data.misc_frozen.saturating_add(acc.data.fee_frozen),
+                    flags: ExtraFlags::default(),
+                };
+
+                // `ensure_upgraded` bumps the consumers if there is a non zero reserved balance and no frozen balance.
+                // https://github.com/paritytech/polkadot-sdk/blob/305d311d5c732fcc4629f3295768f1ed44ef434c/substrate/frame/balances/src/lib.rs#L785
+                let expected_consumers = if acc.data.reserved > 0 && expected_data.frozen == 0 {
+                    acc.consumers + 1
+                } else {
+                    acc.consumers
+                };
+                let expected_acc = frame_system::AccountInfo {
+                    nonce: acc.nonce,
+                    consumers: expected_consumers,
+                    providers: acc.providers,
+                    sufficients: acc.sufficients,
+                    data: expected_data,
+                };
+                expected_account.insert(acc_id, expected_acc);
+            }
+
+            Ok(expected_account.encode())
+        }
+
+        /// Migrates Account storage to the new format, and calls `ensure_upgraded` for them.
+        fn on_runtime_upgrade() -> Weight {
+            // Pull the storage in the previous format into memory
+            let accounts = prev::Account::<Runtime>::iter().collect::<Vec<_>>();
+            log::info!(target: TARGET, "Migrating {} accounts...", accounts.len());
+
+            for (acc_id, acc_info) in accounts.clone().into_iter() {
+                let prev_data = acc_info.clone().data;
+
+                // Move account to new data format
+                let new_data = pallet_balances::AccountData {
+                    free: prev_data.free,
+                    reserved: prev_data.reserved,
+                    frozen: prev_data.misc_frozen.saturating_add(prev_data.fee_frozen),
+                    flags: ExtraFlags::old_logic(),
+                };
+                let new_account = frame_system::AccountInfo {
+                    nonce: acc_info.nonce,
+                    consumers: acc_info.consumers,
+                    providers: acc_info.providers,
+                    sufficients: acc_info.sufficients,
+                    data: new_data,
+                };
+                frame_system::pallet::Account::<Runtime>::insert(acc_id.clone(), new_account);
+
+                // Ensure upgraded
+                pallet_balances::Pallet::<Runtime, ()>::ensure_upgraded(&acc_id);
+            }
+
+            log::info!(target: TARGET, "Migrated {} accounts ✅", accounts.len());
+
+            // R/W not important for solo chain.
+            return <Runtime as frame_system::Config>::DbWeight::get().reads_writes(0u64, 0u64);
```

Comment:
You're better than this, Mr. Rust expert:

```suggestion
            <Runtime as frame_system::Config>::DbWeight::get().reads_writes(0u64, 0u64)
```

Reply:
Huh I'm surprised clippy didn't shout at me about that